File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        
        int count=1;
        int a = nums[0];
        for(int i=1;i<nums.length;i++){
            if(nums[i]!=a){
                count--;
            }
            else{
                count++;
            }
            if(count==0){
                a=nums[i];
                count=1;
            }
        }
        return a;
    }
}

File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        pair<int,int> p; // element and its count
        p = {nums[0],1};

        for(int i=0;i<nums.size();i++){
            if(p.first==nums[i]){
                p.second++;
            }
            else{
                p.second--;
            }
            if(p.second==0){
                p = {nums[i],1};
            }
        }
        return p.first;
    }
};